### DISCUSSION TITLE
How to quote this in task?

### DISCUSSION THREAD

**Discussion by failable** [zed-industries/zed#11208] [Feature Requests]
This config

```json
{
    "label": "Dired",
    "command": "osascript",
    "args": [
      "-e",
      "'activate application \"Emacs\"'",
      "-e",
      "'do shell script \"/usr/local/bin/emacsclient -n -u -e \\\"(dired-jump)\\\"\"'"
    ],
    "use_new_terminal": false,
    "allow_concurrent_runs": false,
    "reveal": "never"
  }
```

works fine in my Zed. 

But I'd like to put some path in the `(dired-jump)` like

```
(dired-jump "...path...here...")
```

where `path` is a path to some directory. This issue is I couldn't find a way to quote / escape the path. The below configs do not work

```
...
 "'do shell script \"/usr/local/bin/emacsclient -n -u -e \\\"(dired-jump \"...path...here...\")\\\"\"'"
...
```

```
...
 "'do shell script \"/usr/local/bin/emacsclient -n -u -e \\\"(dired-jump \\\"...path...here...\\\")\\\"\"'"
...
```

Any help will be appreciated! Thanks!

**Comment by osiewicz** [comment]
Is it any better with recent nightly releases? We've changed how quoting works in https://github.com/zed-industries/zed/pull/10776

**Reply by failable** [reply]
Hi, thanks for the update. The issue is I don't know how to quote it. A workable bash version is

```shell
osascript -e 'do shell script "/usr/local/bin/emacsclient -n -u -e '\''(dired \"/Users/user/Downloads\")'\''"'
```
I don't know how to translate it into Zed task.

I get mess up with the quotes...

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
